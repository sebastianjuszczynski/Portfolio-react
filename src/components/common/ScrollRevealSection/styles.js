import styled from "styled-components";
import { motion } from 'framer-motion';

export const MotionTag = styled(motion.div)`
    display: block;
`;