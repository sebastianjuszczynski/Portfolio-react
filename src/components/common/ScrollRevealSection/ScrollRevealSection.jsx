import { useScrollReveal } from "../../../hooks/useScrollReveal";
import { containerVariants } from "../Animations/animationsVariants";
import { MotionTag } from "./styles";

const ScrollRevealSection = ({ children }) => {
    const { ref, isInView } = useScrollReveal();

    return (
        <MotionTag
            ref={ref}
            variants={containerVariants}
            initial="hidden"
            animate={isInView ? "visible" : "hidden"}
            data-testid="scroll-reveal"
        >
        {children}
        </MotionTag>
    );
};

export default ScrollRevealSection;