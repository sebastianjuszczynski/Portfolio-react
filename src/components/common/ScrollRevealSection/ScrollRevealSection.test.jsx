import { render, screen } from '@testing-library/react';
import ScrollRevealSection from './ScrollRevealSection';

vi.mock('../../../hooks/useScrollReveal', () => ({
    useScrollReveal: () => ({
        ref: { current: null },
        isInView: true
    })
}));

describe('ScrollRevealSection', () => {
    test('renders children correctly', () => {
        render(
            <ScrollRevealSection>
                <p>Test</p>
            </ScrollRevealSection>
        );
        
        expect(screen.getByText(/test/i)).toBeInTheDocument();
    });

    test('renders correct DOM structure when in view', () => {
        const { container } = render(
            <ScrollRevealSection>
                <div>Child</div>
            </ScrollRevealSection>
        );

        expect(container.firstChild).toMatchSnapshot();
    });
});

