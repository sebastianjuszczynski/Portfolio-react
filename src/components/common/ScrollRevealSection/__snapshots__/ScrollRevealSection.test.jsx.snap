// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScrollRevealSection > renders correct DOM structure when in view 1`] = `
.c0 {
  display: block;
}

<div
  class="c0"
  data-testid="scroll-reveal"
>
  <div>
    Child
  </div>
</div>
`;
